src/sentry/static/sentry/app/views/organizationDiscover/result/index.jsx

```diff
@@ -134,17 +138,32 @@ export default class Result extends React.Component {
         {view === 'table' && <Table data={data} query={query} />}
         {view === 'line' && (
           <ChartWrapper>
-            <LineChart series={basicChartData} height={300} tooltip={tooltipOptions} />
+            <LineChart
+              series={basicChartData}
+              height={300}
+              tooltip={tooltipOptions}
+              legend={{data: [query.aggregations[0][2]]}}
+            />
           </ChartWrapper>
         )}
         {view === 'bar' && (
           <ChartWrapper>
-            <BarChart series={basicChartData} height={300} tooltip={tooltipOptions} />
+            <BarChart
+              series={basicChartData}
+              height={300}
+              tooltip={tooltipOptions}
+              legend={{data: [query.aggregations[0][2]]}}
+            />
           </ChartWrapper>
         )}
         {view === 'line-by-day' && (
           <ChartWrapper>
-            <LineChart series={byDayChartData} height={300} tooltip={tooltipOptions} />
+            <LineChart
+              series={byDayChartData}
+              height={300}
+              tooltip={tooltipOptions}
+              legend={legendData ? {data: legendData} : null}
```

Comment:
You can avoid the extra ternarys here and on line 177 by doing this on line 121.
e.g. that would read`? {data: byDayChartData.map(entry => entry.seriesName)}`